src/java_bytecode/java_types.cpp

```diff
@@ -158,7 +162,48 @@ exprt java_bytecode_promotion(const exprt &expr)
     return typecast_exprt(expr, new_type);
 }
 
-typet java_type_from_string(const std::string &src)
+/// find corresponding closing '>' which may be hierarchic
+size_t java_generics_find_closing(const std::string &src, size_t open_pos)
```

Comment:
You might consider unit testing this since this is the kind of thing it is a) easy to unit test and b) easy to make silly mistakes. It also has the added benefit of documenting the "fence post" behaviour (e.g. does it return the position immediately before or after the closing `>`)

You might also consider pulling it into a utility and taking the bracket as an arg as I guess this is quite a common thing to want to do.